src/Tribe/RSVP.php

```diff
@@ -1716,37 +1716,111 @@ public function updated_messages( $messages ) {
+		}
+
+		if ( $rsvp_options[ $previous_order_status ]['decrease_stock_by'] === $rsvp_options[ $attendee_order_status ]['decrease_stock_by'] ) {
+			return false;
+		}
+
+		return array(
+			'previous_stock_size' => $rsvp_options[ $previous_order_status ]['decrease_stock_by'],
+			'attendee_stock_size' => $rsvp_options[ $attendee_order_status ]['decrease_stock_by'],
+		);
+	}
+
+	/**
+	 * Get updated value for stock or sales, based on order status
+	 *
+	 * @param $order_id
+	 * @param $attendee_order_status
+	 * @param $ticket_id
+	 * @param $meta
+	 *
+	 * @return bool|int|mixed get updated value, return false if no need to update
+	 */
+	public function find_updated_sales_or_stock_value( $order_id, $attendee_order_status, $ticket_id, $meta ) {
+		$rsvp_options = $this->tickets_view->get_rsvp_options( null, false );
+
+		$status_stock_sizes = $this->stati_are_different( $order_id, $attendee_order_status );
+
+		if ( empty( $status_stock_sizes ) ) {
+			return false;
+		}
+
+		$diff = $status_stock_sizes['attendee_stock_size'] - $status_stock_sizes['previous_stock_size'];
+
+		if ( 0 === $diff ) {
+			return false;
+		}
+
+		$meta_value = (int) get_post_meta( $ticket_id, $meta, true );
+
+		if ( 'total_sales' === $meta ) {
+			$new_value = $meta_value + $diff;
+		} else {
+			// When we increase sales, we reduce stock
+			$new_value = $meta_value - $diff;
+			// stock can NEVER exceed capacity
+			$capacity = get_post_meta( $ticket_id, '_tribe_ticket_capacity', true );
+			$new_value = ( $new_value > $capacity ) ? $capacity : $new_value;
+		}
+
+		return $new_value;
+	}
+
+	/**
+	 * Updates the product sales and stock if old and new order stati differ in stock size.
+	 *
+	 * @param int    $order_id
+	 * @param string $attendee_order_status
+	 * @param int    $ticket_id
+	 */
+	public function update_sales_and_stock_by_order_status( $order_id, $attendee_order_status, $ticket_id ) {
+		$sales_diff = $this->find_updated_sales_or_stock_value( $order_id, $attendee_order_status, $ticket_id, 'total_sales' );
+
+		// it's all or none here...
+		if ( false === $sales_diff ) {
 			return;
 		}
 
-		$previous_order_status_stock_size = $rsvp_options[ $previous_order_status ]['decrease_stock_by'];
-		$attendee_order_status_stock_size = $rsvp_options[ $attendee_order_status ]['decrease_stock_by'];
+		$stock_diff = $this->find_updated_sales_or_stock_value( $order_id, $attendee_order_status, $ticket_id, '_stock' );
 
-		if ( $previous_order_status_stock_size == $attendee_order_status_stock_size ) {
+		// it's all or none here...
+		if ( false === $stock_diff ) {
 			return;
 		}
 
-		$sales = (int) get_post_meta( $ticket_id, 'total_sales', true );
-		$diff  = $attendee_order_status_stock_size - $previous_order_status_stock_size;
+		// these should NEVER be updated separately - if one goes up the other must go down and vice versa
+		update_post_meta( $ticket_id, 'total_sales', $sales_diff );
+		update_post_meta( $ticket_id, '_stock', $stock_diff );
```

Comment:
Lets return the combined status of these updates. Just to make easier to test this later

Reply:
Hm 🤔 as an array? Something like:
```
return array(
    update_post_meta( $ticket_id, 'total_sales', $sales_diff ),
    update_post_meta( $ticket_id, '_stock', $stock_diff ),
);
```
❓

Reply:
First instinct is boolean since we're returning false (now) earlier?